contract-examples/contracts/OrderBook.sol

```diff
@@ -37,8 +37,12 @@ contract OrderBook is EIP712 {
     function placeOrder(Order memory order, bytes memory signature) external {
         (, bytes32 orderHash) = verifySigner(order, signature);
 
-        // OB_OMBU: Order Must Be Unfilled
-        require(ordersStatus[orderHash] == OrderStatus.Unfilled, "OB_OMBU");
+        // OB_OMBU: Order Must Not Be Unfilled already
+        // ideally, order should not exist in the orderStatus map already
+        // require(ordersStatus[orderHash] == OrderStatus.Unfilled, "OB_OMBU");
+        // bytes32 orderHash = bytes32(hashh);
+        ordersStatus[orderHash] = OrderStatus.Unfilled;
+        // addressStatus[order.trader] = OrderStatus.Cancelled;
```

Comment:
default value for an enum is the 0th entry. so all new orders should have `Unfilled`status

Reply:
But if an order is placed again after being placed successfully, it should fail. But here, the condition will be true since the status for that hash is Unfilled. Maybe we could have a status called NoStatus(index = 0)?

Reply:
yes you are right, we should have 1 more enum to clarify that the order is placed but not fulfilled